resourseleak/TestMessage.java

```diff
@@ -0,0 +1,87 @@
+package resourceleak;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.GZIPOutputStream;
+
+
+public class TestMessage {
+
+    private static Map<String, JsonObject> testMessage1(String directory, String fileNamePrefix) throws IOException {
+        Map<String, JsonObject> modelMeta = new HashMap<>();
+
+        File[] files = new File(directory).listFiles((dir, name) -> name.startsWith(fileNamePrefix));
+        for (File file : files) {
+            try {
+                JsonObject obj = (JsonObject) new JsonParser().parse(new FileReader(file));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource created by the following constructor call: *FileReader*. Currently, there are execution paths that do not contain closure statements, for example, when JsonParser.parse throws an exception. Either a) close the object returned by *FileReader()* in a try-finally block or b) close the resource by declaring the object returned by *FileReader()* in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).